### PR TITLE
Proper merge queue JSON model; add a dummy check

### PR DESCRIPTION
**Pull request by yjbanov**
Get onto the JsonSerializable bandwagon for the `merge_group` data. Add a dummy check so we can start testing MQ <=> Cocoon integration. Failing PRs can be artificially created by adding the string "MQ_FAIL" in the commit message.

### CODE REVIEW

File: app_dart/lib/src/model/github/checks.dart [hunk 1]
```diff
@@ -52,3 +52,96 @@ class CheckRun {
...
+
+  /// The commit SHA generated for the candidate PR to be merged.
+  ///
+  /// This is the SHA that must be checked by the CI.
+  final String headSha;
+
+  /// The ref to the read-only branch generated by Github for the respective PR.
+  final String headRef;
+
+  /// The SHA of the parent commit of the [headSha] commit.
```
Comment (jtmcdole) [inline app_dart/lib/src/model/github/checks.dart:108]: Maybe:
```
///
/// This is the SHA head of a merge queue protect branch
```

File: app_dart/lib/src/request_handlers/github/webhook_subscription.dart [hunk 2]
```diff
@@ -233,8 +234,10 @@ class GithubWebhookSubscription extends SubscriptionHandler {
       throw BadRequestException('Malformed merge_group request:\n$rawRequest');
     }
 
-    final eventAction = request['action'] as String;
-    final headSha = (request['merge_group'] as Map<String, Object?>)['head_sha'] as String;
+    final mergeGroupEvent = MergeGroupEvent.fromJson(request);
```
Comment (jtmcdole) [inline app_dart/lib/src/request_handlers/github/webhook_subscription.dart:237]: fun:
  final MergeGroupEvent(:mergeGroup, :action, :headSha) = MergeGroupEvent.fromJson(request);

but you don't use eventAction?

**Review by jtmcdole — APPROVED** [review]

File: app_dart/lib/src/request_handlers/github/webhook_subscription.dart (hunk 2 above)
Comment (jtmcdole) [reply app_dart/lib/src/request_handlers/github/webhook_subscription.dart:237]: ignore the second part; eventAction is used in the zippie

Comment (yjbanov) [reply app_dart/lib/src/request_handlers/github/webhook_subscription.dart:237]: Done.

File: app_dart/lib/src/model/github/checks.dart (hunk 1 above)
Comment (yjbanov) [reply app_dart/lib/src/model/github/checks.dart]: Used github docs verbatim for now. We can clarify when we know more about what this is.

**Review by jtmcdole — APPROVED** [review]
still LGTM

**Pull request merged**